test/unit/deployment.bats

```diff
@@ -178,6 +178,75 @@ load _helpers
     [ "${actual}" = "true" ]
 }
 
+# podSecurityContext
+@test "controller/Deployment: controller.podSecurityContext set by default" {
+  cd `chart_dir`
+  local object=$(helm template \
+      -s templates/deployment.yaml  \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec.securityContext | select(documentIndex == 1)' | tee /dev/stderr)
+
+   local actual=$(echo "$object" | yq '. | length' | tee /dev/stderr)
+   [ "${actual}" = "1" ]
+   actual=$(echo "$object" | yq '.runAsNonRoot' | tee /dev/stderr)
+   [ "${actual}" = "true" ]
+
+}
+
+@test "controller/Deployment: controller.podSecurityContext can be set" {
+  cd `chart_dir`
+  local object=$(helm template \
+      -s templates/deployment.yaml  \
+      --set 'controller.podSecurityContext.runAsGroup=2000' \
+      --set 'controller.podSecurityContext.runAsUser=2000' \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec.securityContext | select(documentIndex == 1)' | tee /dev/stderr)
+
+   local actual=$(echo "$object" | yq '. | length' | tee /dev/stderr)
+   [ "${actual}" = "3" ]
+   actual=$(echo "$object" | yq '.runAsGroup' | tee /dev/stderr)
+   [ "${actual}" = '2000' ]
+   actual=$(echo "$object" | yq '.runAsUser'| tee /dev/stderr)
+   [ "${actual}" = '2000' ]
+
+}
+
+# securityContext
+@test "controller/Deployment: controller.{manager,kube-rbac-proxy}.securityContext set by default" {
+  cd `chart_dir`
+  local object=$(helm template \
+      -s templates/deployment.yaml  \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec | select(documentIndex == 1)' | tee /dev/stderr)
+
+   local actual=$(echo "$object" | yq '.containers[0].securityContext.allowPrivilegeEscalation' | tee /dev/stderr)
+    [ "${actual}" = "false" ]
+
+   actual=$(echo "$object" | yq '.containers[1].securityContext.allowPrivilegeEscalation' | tee /dev/stderr)
+    [ "${actual}" = "false" ]
+
+}
+
+@test "controller/Deployment: controller.{manager,kube-rbac-proxy}.securityContext can be set" {
+  cd `chart_dir`
+  local object=$(helm template \
+      -s templates/deployment.yaml  \
+      --set 'controller.kubeRbacProxy.securityContext.allowPrivilegeEscalation=true' \
+      --set 'controller.manager.securityContext.allowPrivilegeEscalation=true' \
```

Comment:
This test fails as the names have changed (I also fixed this up in f6a5aa8b98aa27b237b0976a168a7c3b93b73f5c):

```suggestion
      --set 'controller.securityContext.allowPrivilegeEscalation=true' \
```

Sorry if it's kinda difficult to get going with the tests. If you'd like to be able to run the tests locally, you can install `bats` from [this homebrew package](https://formulae.brew.sh/formula/bats-core#default), and then from the root of the repo run:

```bash
PATH="$(pwd)/scripts:$PATH" bats -f "ecurityContext" test/unit/
```

But I'm also happy to just continue triggering CI for you if you'd rather not.